examples/k8s/hz-go-service/service.go

```diff
@@ -0,0 +1,347 @@
+		if err != nil {
+			http.Error(rw, "Invalid request body", http.StatusBadRequest)
+			return
+		}
+		var rsp []byte
+		pmap, err := hzc.GetMap(ctx, mapName)
+		if err != nil {
+			http.Error(rw, "Hazelcast cannot return a map", http.StatusInternalServerError)
+			return
+		}
+		_, err = pmap.Put(ctx, key, reqBody.Value)
+		if err != nil {
+			http.Error(rw, fmt.Sprintf("Hazelcast cannot put %v value to map", reqBody.Value), http.StatusInternalServerError)
+			return
+		}
+		successInfo := Info{
+			Msg: fmt.Sprintf("(Key: %s, Value: %v) has been put to Map: %s successfully",
+				key, reqBody.Value, mapName),
+		}
+		rsp, err = json.Marshal(successInfo)
+		if err != nil {
+			http.Error(rw, fmt.Sprintf("Marshal cannot work properly for non exist map put, rsp: %v", successInfo), http.StatusInternalServerError)
+			return
+		}
+		_, err = rw.Write(rsp)
+		if err != nil {
+			http.Error(rw, fmt.Sprintf("Response cannot be written properly"), http.StatusInternalServerError)
+			return
+		}
+	}
+}
+
+// isContentTypeJson determine whether content-type of the request header is in json type.
+func isContentTypeJson(r *http.Request) bool {
+	contentType := r.Header.Get("Content-type")
+	if contentType == "" {
+		return false
+	}
+	if contentType != "application/json" {
+		return false
+	}
+	return true
+}
+
+// healthHandler handlers for k8s pod health.
+func healthHandler(rw http.ResponseWriter, _ *http.Request) {
+	rw.WriteHeader(http.StatusOK)
+}
+
+// readinessHandler handlers for k8s pod readiness.
+func readinessHandler(rw http.ResponseWriter, _ *http.Request) {
+	rw.WriteHeader(http.StatusOK)
+}
+
+// newDefaultServiceConfig returns new config.
+func newDefaultServiceConfig() *ServiceConfig {
+	// default config properties
+	return &ServiceConfig{
+		ServiceName: "hz-go-service",
+		Port:        8080,
+		Timeout:     5 * time.Second,
+	}
+}
+
+// newInMemoryHzMap creates a sample map and put example entries on it.
+func newInMemoryHzMap(ctx context.Context, c *hazelcast.Client) (*InMemoryHzStorage, error) {
+	entries := []types.Entry{
+		{Key: "key1", Value: "value1"},
+		{Key: "key2", Value: "value2"},
+		{Key: "key3", Value: "value3"},
+	}
+	newMap, err := c.GetMap(ctx, sampleMap)
+	err = newMap.PutAll(ctx, entries...)
+	if err != nil {
+		return nil, err
+	}
+	return &InMemoryHzStorage{myMap: newMap}, nil
+}
+
+// newServer initiates new server instance according to given router and config.
+func newServer(router *mux.Router, config *ServiceConfig) *http.Server {
+	return &http.Server{
+		Handler:      router,
+		Addr:         fmt.Sprintf(":%d", config.Port),
+		ReadTimeout:  config.Timeout,
+		WriteTimeout: config.Timeout,
+	}
+}
+
+// newRouter returns a new mux router includes certain handlers.
+func newRouter() *mux.Router {
+	// creates a server
+	router := mux.NewRouter()
+	// health checks for kubernetes
+	router.HandleFunc("/health", healthHandler)
+	router.HandleFunc("/readiness", readinessHandler)
```

Comment:
```suggestion
       okHandler := func (rw http.ResponseWriter, _ *http.Request) {
		rw.WriteHeader(http.StatusOK)
	}
	router.HandleFunc("/health", okHandler)
	router.HandleFunc("/readiness", okHandler)
```
I think we can get rid of the identical named functions. Wdyt?

Reply:
Good idea, [9c4baec](https://github.com/hazelcast/hazelcast-go-client/pull/743/commits/9c4baecc96452e673ce3249504f10ce2efa3d810).